heat/core/manipulations.py

```diff
@@ -179,3 +180,170 @@ def squeeze(x, axis=None):
+
+    # Share and gather the results with the other processes
+    uniques = torch.tensor([lres.shape[0]]).to(torch.int32)
+    uniques_buf = torch.empty((a.comm.Get_size(), ), dtype=torch.int32)
+    a.comm.Allgather(uniques, uniques_buf)
+
+    split = None
+    is_split = None
+
+    if axis is None or axis is a.split:
+        # Local results can now just be added together
+        if axis is None:
+            # One dimensional vectors can't be distributed -> no split
+            output_dim = [uniques_buf.sum().item()]
+            recv_axis = 0
+        else:
+            output_dim = list(lres.shape)
+            output_dim[0] = uniques_buf.sum().item()
+            recv_axis = a.split
+
+            # Result will be split along the same axis as a
+            split = a.split
+
+        # Gather all unique vectors
+        counts = list(uniques_buf.tolist())
+        displs = list([0] + uniques_buf.cumsum(0).tolist()[:-1])
+        gres_buf = torch.empty(output_dim, dtype=a.dtype.torch_type())
+        a.comm.Allgatherv(lres, (gres_buf, counts, displs,), axis=recv_axis, recv_axis=0)
+
+        if return_inverse:
+            # Prepare some information to generated the inverse indices list
+            if axis is None:
+                inverse_indices = inverse_pos
+            else:
+                avg_len = a.gshape[a.split] // a.comm.Get_size()
+                rem = a.gshape[a.split] % a.comm.Get_size()
+
+                # Share the local reverse indices with other processes
+                counts = [avg_len] * a.comm.Get_size()
+                add_vec = [1] * rem + [0] * (a.comm.Get_size() - rem)
+                inverse_counts = [sum(x) for x in zip(counts, add_vec)]
+                inverse_displs = [0] + list(np.cumsum(inverse_counts[:-1]))
+                inverse_dim = list(inverse_pos.shape)
+                inverse_dim[0] = a.gshape[0]
+                inverse_buf = torch.empty(inverse_dim, dtype=inverse_pos.dtype)
+                a.comm.Allgatherv(inverse_pos, (inverse_buf, inverse_counts, inverse_displs))
+
+        # Run unique a second time for
+        gres = torch.unique(gres_buf, sorted=sorted, return_inverse=return_inverse, dim=unique_axis)
+
+        if return_inverse and axis is not None:
+            # Use the previously gathered information to generate global inverse_indices
+            g_inverse = gres[1]
+            gres = gres[0]
+            inverse_indices = torch.zeros_like(inverse_buf)
+            steps = displs + [None]
+
+            # Algorithm that creates the correct list for the reverse_indices
+            for i in range(len(steps) - 1):
+                begin = steps[i]
+                end = steps[i + 1]
+                for num, x in enumerate(inverse_buf[begin: end]):
+                    inverse_indices[begin + num] = g_inverse[begin + x]
+
+    else:
+        max_uniques, max_pos = uniques_buf.max(0)
+
+        # find indices of vectors
+        if a.comm.Get_rank() is max_pos.item():
```

Comment:
`is` is the wrong operator here. It checks for equality and identity, equality is sufficient and what you actually want